### ISSUE TITLE
Auto-Formatting w/ Ruff not working

### ISSUE THREAD

**Issue by jjfantini** [getcursor/cursor#841]
**Is your feature request related to a problem? Please describe.**
When using Ruff or Black Formatter, the file does not automatically format on a file save or on type. I have the same settings in VSCode and Cursor. When I have extra whitespace underneath an import and save the file in VSCode, the file auto formats and removes the whitespace, but this does NOT happen in Cursor.

**Describe the solution you'd like**
Auto-formatting when using Ruff or Black.

**Additional context**
I can see the output of Ruff in vscode, code is sent and executed, but this does not happen n Cursor. This is giving me a headache, I miss my old flow! I didn't have to worry about any of this :)

**Comment by sagarspatil** [comment]
+1 Does not work with Black formatter either.

**Comment by jjfantini** [comment]
Any progress with this?

**Comment by jjfantini** [comment]
I constantly get:

`FormattingExtension 'Black Formatter' is configured as formatter but it cannot format 'Python'-files`

**Comment by jjfantini** [comment]
Auto-formatting is essential. I have to have VScode open as well to format my files as I write them. I have gotten use to the flow of RUff auto-formatting as I type, so not having this brings the IDE back a few years and makes it un-usable. 

It must be something with the Python Language Server?

Cursor Ruff: `Running Ruff with: c:\Users\jjfan\.cursor\extensions\charliermarsh.ruff-2023.38.0-win32-x64\bundled\libs\bin\ruff.exe`

VSCode Ruff: `Running Ruff with: c:\Users\jjfan\OneDrive\Documents\VS_WD\humblData_v2\obb\Scripts\ruff.exe`
- VSCode uses the conda environment installation of Ruff

The settings for Ruff are the same in both IDE's

**Comment by jjfantini** [comment]
![ruff-error](https://github.com/getcursor/cursor/assets/63276164/8f57fdaa-c067-45a1-b6a2-1c91236ed215)

I have made the path the same as the VSCode path, as you can see in the screenshot, but I am still not able to format python files...

**Comment by sagarspatil** [comment]
The problem was with my user settings file. Go here: C:\Users\sagar\AppData\Roaming\Cursor\User and make necessary changes.
Here's my settings for reference.

```
{
    "[python]": {
        "editor.formatOnType": true,
        "editor.defaultFormatter": "ms-python.black-formatter"
    },
    "git.autofetch": true,
    "git.enableSmartCommit": true,
    "git.confirmSync": false,
    "editor.formatOnSave": true,
    "editor.formatOnPaste": true,
    "black-formatter.args": [
        "--line-length",
        "79"
    ],
    "workbench.activityBar.orientation": "vertical",
    "code-runner.runInTerminal": true,
    "workbench.editorAssociations": {
        "*.py": "default"
    },
    "workbench.tree.indent": 15,
    "workbench.tree.renderIndentGuides": "always",
    "workbench.colorCustomizations": {
        "tree.indentGuidesStroke": "#05ef3c"
    },
    "editor.wordWrap": "on",
    "terminal.integrated.allowMnemonics": true,
    "terminal.integrated.sendKeybindingsToShell": true
}
```

**Comment by jjfantini** [comment]
BLack does seem to be working, but Ruff is still not working

**Comment by jjfantini** [comment]
There is no option when configuring Default Formatters to choose Ruff. You can obviously type it in the settings.json, but it doesn't show up in the IDE GUI when selecting between Black and Python. @truell20

**Comment by jjfantini** [comment]
It seems like I still can't get Ruff or Black to format a file on save or on type?

I would like to use the auto-formatting feature of these tools. I am just trying to have a line break when the line length is over the threshold.

Can you address on this? It works on VSCode :)

@truell20

**Comment by jjfantini** [comment]
@sagarspatil is auto-formatting working for you?

**Comment by sagarspatil** [comment]
> @sagarspatil is auto-formatting working for you?

Yes. Although, I now use ruff.

**Comment by tarikozket** [comment]
this has been fixed; the issue could be closed.

**Comment by realjoenguyen** [comment]
@tarikozket ruff doesn't work in my Cursor

**Comment by ashwinb** [comment]
It seemed like things were just working but suddenly stopped working since the latest update perhaps?

**Comment by sagarspatil** [comment]
Yep. Spent an hour trying to debug. I’m on 0.46. Definitely something is up with Cursor.

**Comment by Joker666** [comment]
Not working for me either.

**Comment by westonpace** [comment]
I encountered this with 0.46.8.  The issue went away when upgrading to 0.47.8.
